libshaderc/src/shaderc.cc

```diff
@@ -85,21 +85,55 @@ struct {
 
 std::mutex compile_mutex;  // Guards shaderc_compile_*.
 
-// Rejects #include directives.
-class ForbidInclude : public shaderc_util::CountingIncluder {
+class InternalFileIncluder : public shaderc_util::CountingIncluder {
  private:
-  // Returns empty contents and an error message.  See base-class version.
+  bool AreValidCallbacks() const {
+    return GetIncluderResponse_ != nullptr &&
+           ReleaseIncluderResponse_ != nullptr;
+  }
+
+  // Find filename in search path and returns its contents.
   std::pair<std::string, std::string> include_delegate(
       const char* filename) const override {
```

Comment:
Should have public first, then protected, private.
https://google.github.io/styleguide/cppguide.html#Declaration_Order